### ISSUE TITLE
custom plugin options/settings page visible only to custom role's dashaboard

### ISSUE THREAD

**Issue by steveadamsfixedit** [vijayhardaha/wordpress-plugin-boilerplate#1]
I'm building a plugin based on the boilerplate that contains analytics data on virtual events attended by logged in users. The entire system is restricted to a hierarchy of users from the top down as follows: Admins, Moderators, Presenters, Attendees. I've granted Moderators access to the WP dashboard where I want to display a 'settings/options' page where they can see the activity analytics data of the attendees for only the events that are under their moderation control. Admins should be able to see all event analytics for all events tied to all moderators. 
   Currently, I've constructed this page in my custom_plugin/admin/partials/custom_plugin-events.php file and it is displaying all the activity at an administrator level. Furthermore, I've constructed the custom post type for the events and the custom roles with access. The moderator can see the custom post type in their dashboard.
    My question is threefold:

1) How do I make the event analytics page at the wp-admin/admin.php?page=custom_plugin-events url visible in the moderator's dashboard?
2) I'm presently using some conditional logic in this admin/partial page that basically says if user_can=admin show this data and if user_can=moderator show this other data. Is that the correct and recommended approach? 
3) If I'm requiring all these users to be logged in and applied to one custom role or another...is the custom_plugin/admin directory  the correct place to implement this? or would it be better in custom_plugin/public?

**Comment by vijayhardaha** [comment]
Hey @steveadamsfixedit 

1. admin pages are registered via [add_menu_page](https://developer.wordpress.org/reference/functions/add_menu_page/) & [add_submenu_page](https://developer.wordpress.org/reference/functions/add_submenu_page/) functions that you can find in `class-custom-plugin-admin.php` file. You can read about those functions in WordPress documentation how they work and then you can easily register your own page on the desired location.
2. for the conditional logic it's up to you how you wanna manage that. as you said you're working with user roles then you can get the check that like this
```
$user = wp_get_current_user();
if ( in_array( 'author', (array) $user->roles ) ) {
    //The user has the "author" role
}
```
3. `custom_plugin/admin` directory is just to keep the plugin structure clean and understandable, it's not restricted by boilerplate. if your code should only execute in backend/admin then you can manage that file in the `admin` folder or if your code should be executed everywhere then you can keep that outside the `admin` folder. if you'll check the `class-custom-plugin.php` file you'll find this line `if ( $this->is_request( 'admin' ) )` here I am loading those files that should be only executed in the backend. so you get an idea of how you have to load files. admin folder is just a structure, if you'll create any new file in the admin folder you'll have to load that file in that above function where you'll find `if ( $this->is_request( 'admin' ) )` so it's up to how you wanna load your files.

In summary, this boilerplate just provide the basic structure and some pre-built functions that require & useful for the plugin development then the rest is up to you for code.

Thanks
